lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -716,6 +716,54 @@ defmodule Mix.Tasks.TestTest do
     end
   end
 
+  describe "--dry-run" do
+    test "works with --stale" do
+      in_fixture("test_stale", fn ->
+        File.write!("test/dry_run_one_test_stale.exs", """
+        defmodule DryRunOneTest do
+          use ExUnit.Case
+
+          test "new test" do
+            assert true
+          end
+        end
+        """)
+
+        File.write!("test/dry_run_two_test_stale.exs", """
+        defmodule DryRunTwoTest do
+          use ExUnit.Case
+
+          @tag :skip
+          test "skipped test" do
+            assert true
+          end
+        end
+        """)
+
+        output = mix(["test", "--dry-run", "--stale"])
+
+        assert output =~ "Tests that would be executed:"
+        assert output =~ "test/a_test_stale.exs:4"
+        assert output =~ "test/b_test_stale.exs:4"
+        assert output =~ "test/dry_run_one_test_stale.exs:4"
+        refute output =~ "test/dry_run_two_test_stale.exs:5"
+        assert output =~ "1 test, 0 failures, 1 skipped"
+      end)
+    end
+
+    test "works with --failed" do
+      in_fixture("test_failed", fn ->
+        _initial_run = mix(["test"])
+        output = mix(["test", "--dry-run", "--failed"])
```

Comment:
And then let's run this command again to make sure none of the tests were marked as passing. :)